refactor: remove query by events for x/gov deposits (#9519)
cosmos/cosmos-sdk: x/gov/spec/01_concepts.md

### Deposit refund and burn

> When a proposal finalized, the coins from the deposit are either refunded or burned, according to the final tally of the proposal:

Reviewer: ```suggestion
When a proposal is finalized, the coins from the deposit are either refunded or burned, according to the final tally of the proposal:
```

> - If the proposal is approved or if it's rejected but _not_ vetoed, deposits will automatically be refunded to their respective depositor (transferred from the governance `ModuleAccount`) and keeps the deposits information in state.

Reviewer: ```suggestion
- If the proposal is approved or rejected but _not_ vetoed, each deposit will be automatically refunded to its respective depositor (transferred from the governance `ModuleAccount`) and the deposit information will remain in state.
```

> - When the proposal is vetoed with a supermajority, deposits will be burned from the governance `ModuleAccount` along with that proposal info and it's deposits info get removed from the state.

Reviewer: ```suggestion
- When the proposal is vetoed with a supermajority, deposits will be burned from the governance `ModuleAccount` and the proposal information along with its deposit information will be removed from state.
```

### Deposit refund and burn

When a proposal is finalized, the coins from the deposit are either refunded or burned, according to the final tally of the proposal:

> - If the proposal is approved or rejected but _not_ vetoed, each deposit will be automatically refunded to its respective depositor (transferred from the governance `ModuleAccount`). All the deposits will be removed from state after refunding to the depositors.
> - When the proposal is vetoed with a supermajority, deposits will be burned from the governance `ModuleAccount` and the proposal information along with its deposit information will be removed from state.

Reviewer 2: ```suggestion
- If the proposal is approved or rejected but _not_ vetoed, each deposit will be automatically refunded to its respective depositor (transferred from the governance `ModuleAccount`). 
- When the proposal is vetoed with a supermajority, deposits will be burned from the governance `ModuleAccount` and the proposal information along with its deposit information will be removed from state.
- All refunded or burned deposits are removed from the state. Events are issued when burning or refunding a deposit.
```
